pageserver/src/page_service.rs

```diff
@@ -990,27 +991,64 @@ impl postgres_backend_async::Handler for PageServerHandler {
     }
 }
 
+#[derive(thiserror::Error, Debug)]
+enum GetActiveTenantError {
+    #[error(
+        "Timed out waiting {wait_time:?} for tenant active state. Latest state: {latest_state:?}"
+    )]
+    WaitForActiveTimeout {
+        latest_state: TenantState,
+        wait_time: Duration,
+    },
+    #[error(transparent)]
+    Other(#[from] anyhow::Error),
+}
+
+impl From<GetActiveTenantError> for QueryError {
+    fn from(e: GetActiveTenantError) -> Self {
+        match e {
+            GetActiveTenantError::WaitForActiveTimeout { .. } => QueryError::Disconnected(
+                ConnectionError::Socket(io::Error::new(io::ErrorKind::TimedOut, e.to_string())),
+            ),
```

Comment:
Is it this conversion which prohibits us stacktrace collection and logging..?

Reply:
Yes, since `io::Error` does not have any backtraces in it now.

Reply:
Yeah, just wondering if there should be a comment. For me it always feels wrong to create new `std::io::Error`'s but perhaps this one is good.

Reply:
I'd say, `std::io::Error`'s should be created judiciously, it's not wrong to create them per se with their current design.
One particular pitfall is https://github.com/neondatabase/neon/pull/3254 , when there's 0 path in the 

> Directory not empty (os error 39)

error from `std::io::Error` bubbled up.

Not quite sure what to comment here, since it's basically the type level changes that I did here: it's either `Other(anyhow::Error)` or a certain type of an error, and I've used a more specific type here.
But would be glad to insert anything proposed.

Reply:
Perhaps no comment is needed after all. Resolving.